README.md

```diff
@@ -14,6 +14,12 @@ Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
 
 Business logic "join" module to provide simple access to all user-centric data.
 
+## Optional dependencies
+
+This module has optional interface dependencies in its [module descriptor](descriptors/ModuleDescriptor-template.json). This allows operators to choose which dependent modules to install for the desired features of this module, and does not require that all dependent modules be installed in every FOLIO instance. For example, operators can install mod-users-bl with the modules designated in the `requires` array, and perform a successful login through the UI. However, unless the dependent modules of the `service-points` and `service-points-users` interfaces are installed in the FOLIO instance, service points will not be returned in the login response. The login response will not produce an HTTP error response however. Instead a partial login response (a partial `CompositeUser` object) is returned without the service points included. Okapi will however log the 404 for the call to the service points endpoint letting the operator know that a dependency is missing for this feature.
+
+The same is true of other optional dependencies. If operators need password reset, they should install the dependent modules of the `password-validator` and `notify` interfaces. Like the example above with service points, the password reset operation will not produce an HTTP error response if `notify` isn't installed. Instead the notification will not be sent, and an error is logged by this module and okapi.
+
```

Comment:
This information is missing from the RAML or JSON files in the /raml/ directory and therefore won't end up in the official API documentation https://s3.amazonaws.com/foliodocs/api/mod-users-bl/r/mod-users-bl.html that UI developers, other developers and sysOps use.

Reply:
So module  documentation is not official.  That's a new one

Reply:
I think the challenge that Julian is referring to is the distinction between interface documentation and implementation (module) documentation. And that folks like UI developers use the interface documentation and thus don't tend to refer to the implementation documentation like a module readme.